pysolarfocus/__init__.py

```diff
@@ -308,35 +34,22 @@ def __init__(self,
                  port:int=PORT,
                  slave_id:int=SLAVE_ID):
         """Initialize Solarfocus communication."""
-        assert heating_circuit_count >= 1 and heating_circuit_count < 9, "Heating circuit count must be between 1 and 8"
-        assert buffer_count >= 1 and buffer_count < 5, "Buffer count must be between 1 and 4"
-        assert boiler_count >= 1 and boiler_count < 5, "Boiler count must be between 1 and 4"
+        assert heating_circuit_count >= 0 and heating_circuit_count < 9, "Heating circuit count must be between 0 and 8"
+        assert buffer_count >= 0 and buffer_count < 5, "Buffer count must be between 0 and 4"
+        assert boiler_count >= 0 and boiler_count < 5, "Boiler count must be between 0 and 4"
         
         self.__conn = ModbusConnector(ip,port,slave_id)
         self.__factory = ComponentFactory(self.__conn)
         #Lists of components
-        self.heating_circuits = self.__factory.heating_circuit(system,heating_circuit_count)
-        self.boilers = self.__factory.boiler(system,boiler_count)
-        self.buffers = self.__factory.buffer(system,buffer_count)
+        self.heating_circuits = self.__factory.heating_circuit(system,heating_circuit_count) if heating_circuit_count > 0 else []
+        self.boilers = self.__factory.boiler(system,boiler_count) if boiler_count > 0 else []
+        self.buffers = self.__factory.buffer(system,buffer_count) if buffer_count > 0 else []
```

Comment:
The count check and `0` handling should probably be done by the factory methodes.